arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/AccountSnapshot.java

```diff
@@ -77,12 +80,16 @@ private static AccountSnapshot fromArguments(
 
     final Account account = worldView.get(address);
     if (account != null) {
+
       return new AccountSnapshot(
           account.getAddress(),
           account.getNonce(),
           account.getBalance(),
           warmth,
-          new Bytecode(account.getCode()),
+          // new Bytecode(account.getCode()),
+          deploymentInfo.getDeploymentStatus(address)
+              ? new Bytecode(deploymentInfo.getInitializationCode(address))
+              : new Bytecode(account.getCode()),
```

Comment:
We provide either the deployed bytecode (if available) or the initialization code.